core/server/web/shared/middlewares/image/handle-image-sizes.js

```diff
@@ -0,0 +1,59 @@
+const path = require('path');
+const sharp = require('sharp');
+const storage = require('../../../../adapters/storage');
+const activeTheme = require('../../../../services/themes/active');
+
+const SIZE_PATH_REGEX = /^\/size\/([^/]+)\//;
+
+module.exports = function (req, res, next) {
+    if (!SIZE_PATH_REGEX.test(req.url)) {
+        return next();
+    }
+
+    const imageSizes = activeTheme.get().config('image_sizes');
+    const imageDimensions = Object.keys(imageSizes).reduce((dimensions, size) => {
+        const {width, height} = imageSizes[size];
+        const dimension = (width ? 'w' + width : '') + (height ? 'h' + height : '');
+        return Object.assign({
+            [dimension]: imageSizes[size]
+        }, dimensions);
+    }, {});
+
+    const [sizeImageDir, requestedDimension] = req.url.match(SIZE_PATH_REGEX);
+
+    // CASE: unknown size or missing size config
+    const imageDimensionConfig = imageDimensions[requestedDimension];
+    if (!imageDimensionConfig || (!imageDimensionConfig.width && !imageDimensionConfig.height)) {
```

Comment:
Just wondering about support for the following case. Having defined only this image size in config:
```
            "small": {
                "width": 300,
                "height": 200
            },
```

With the current implementation, we will redirect to the original image when both width&height don't match. But would it make sense to use config above for when we only specify one of the parameters in the request, e.g: `/content/images/size/w300`? When there are more configs defined with such width we could default to the first one that comes up. Just an idea that wanted to bounce off :thinking:

Reply:
I'm pretty sure that's as specced, most of the time these image sizes are going to be created via a handlebars helper ala `{{img_url profile_img size="something"}}` 

pinging @JohnONolan - do you have any thoughts on this?

Reply:
The URL and the config should be a 1:1 match, this is a fairly fundamental assumption for future proofing, caching, and so on. We should not do any guessing or falling back.